internal/services/netapp/netapp_volume_helper.go

```diff
@@ -0,0 +1,978 @@
+				HasRootAccess:       utils.Bool(rootAccessEnabled),
+				Kerberos5ReadOnly:   utils.Bool(kerberos5ReadOnly),
+				Kerberos5ReadWrite:  utils.Bool(kerberos5ReadWrite),
+				Kerberos5iReadOnly:  utils.Bool(kerberos5iReadOnly),
+				Kerberos5iReadWrite: utils.Bool(kerberos5iReadWrite),
+				Kerberos5pReadOnly:  utils.Bool(kerberos5pReadOnly),
+				Kerberos5pReadWrite: utils.Bool(kerberos5pReadWrite),
+			}
+
+			results = append(results, result)
+		}
+	}
+
+	return &volumes.VolumePatchPropertiesExportPolicy{
+		Rules: &results,
+	}
+}
+
+func expandNetAppVolumeDataProtectionReplication(input []interface{}) *volumes.VolumePropertiesDataProtection {
+	if len(input) == 0 || input[0] == nil {
+		return &volumes.VolumePropertiesDataProtection{}
+	}
+
+	replicationObject := volumes.ReplicationObject{}
+
+	replicationRaw := input[0].(map[string]interface{})
+
+	if v, ok := replicationRaw["endpoint_type"]; ok {
+		endpointType := volumes.EndpointType(v.(string))
+		replicationObject.EndpointType = &endpointType
+	}
+	if v, ok := replicationRaw["remote_volume_location"]; ok {
+		replicationObject.RemoteVolumeRegion = utils.String(v.(string))
+	}
+	if v, ok := replicationRaw["remote_volume_resource_id"]; ok {
+		replicationObject.RemoteVolumeResourceId = v.(string)
+	}
+	if v, ok := replicationRaw["replication_frequency"]; ok {
+		replicationSchedule := volumes.ReplicationSchedule(translateTFSchedule(v.(string)))
+		replicationObject.ReplicationSchedule = &replicationSchedule
+	}
+
+	return &volumes.VolumePropertiesDataProtection{
+		Replication: &replicationObject,
+	}
+}
+
+func expandNetAppVolumeDataProtectionSnapshotPolicy(input []interface{}) *volumes.VolumePropertiesDataProtection {
+	if len(input) == 0 || input[0] == nil {
+		return &volumes.VolumePropertiesDataProtection{}
+	}
+
+	snapshotObject := volumes.VolumeSnapshotProperties{}
+
+	snapshotRaw := input[0].(map[string]interface{})
+
+	if v, ok := snapshotRaw["snapshot_policy_id"]; ok {
+		snapshotObject.SnapshotPolicyId = utils.String(v.(string))
+	}
+
+	return &volumes.VolumePropertiesDataProtection{
+		Snapshot: &snapshotObject,
+	}
+}
+
+func expandNetAppVolumeDataProtectionSnapshotPolicyPatch(input []interface{}) *volumes.VolumePatchPropertiesDataProtection {
+	if len(input) == 0 || input[0] == nil {
+		return &volumes.VolumePatchPropertiesDataProtection{}
+	}
+
+	snapshotObject := volumes.VolumeSnapshotProperties{}
+
+	snapshotRaw := input[0].(map[string]interface{})
+
+	if v, ok := snapshotRaw["snapshot_policy_id"]; ok {
+		snapshotObject.SnapshotPolicyId = utils.String(v.(string))
+	}
+
+	return &volumes.VolumePatchPropertiesDataProtection{
+		Snapshot: &snapshotObject,
+	}
+}
+
+func flattenNetAppVolumeGroupVolumes(ctx context.Context, input *[]volumegroups.VolumeGroupVolumeProperties, metadata sdk.ResourceMetaData) ([]NetAppVolumeGroupVolume, error) {
+	results := make([]NetAppVolumeGroupVolume, 0)
+
+	if len(*input) == 0 || input == nil {
```

Comment:
This is a potential crash and should be swapped as we are casting to a potential nil before checking to see if it's nil
```suggestion
	if input == nil || len(*input) == 0 {
```

Reply:
This is resolved.